### ISSUE TITLE
Download of fasta file from speciality genes fails if more than one page in result set

### ISSUE THREAD

**Issue by fbjornlund** [PATRIC3/patric3_website#1717]
If there is more than one page in the result set of speciality genes then the download of fasta files fails resulting in a 0 byte file. It's also not possible to view the fasta sequences.

This was first discovered trying the enhancement in
https://github.com/PATRIC3/patric3_website/issues/1452

The problem can be reproduced the result set from, select all then try to download fasta (fails). 
https://www.patricbrc.org/view/Genome/83332.12#view_tab=specialtyGenes&filter=eq(source,PATRIC)

If I choose to view Fasta data, then a new window opens saying "loading FASTA data ...", then just showing a blank page. 

If I then from the blank FASTA view page attempts to download FASTA data, 
(url
https://p3.theseed.org/services/data_api/genome_feature/?in(feature_id,(PATRIC.83332.12.NC_000962.CDS.2809936.2810928.rev,PATRIC.83332.12.NC_000962.CDS.2810993.2812066.rev,PATRIC.83332.12.NC_000962.CDS.4063254.4063856.rev,PATRIC.83332.12.NC_000962.CDS.1939599.1940291.fwd,PATRIC.83332.12.NC_000962.CDS.2412119.2413351.rev,PATRIC.83332.12.NC_000962.CDS.1674202.1675011.fwd,PATRIC.83332.12.NC_000962.CDS.2745314.2745736.rev,PATRIC.83332.12.NC_000962.CDS.3882834.3883442.fwd,PATRIC.83332.12.NC_000962.CDS.2516787.2517695.fwd,PATRIC.83332.12.NC_000962.CDS.4151180.4152196.rev,PATRIC.83332.12.NC_000962.CDS.7302.9818.fwd,PATRIC.83332.12.NC_000962.CDS.2710351.2711319.rev,PATRIC.83332.12.NC_000962.CDS.1938146.1938841.fwd,PATRIC.83332.12.NC_000962.CDS.1058260.1060575.fwd,PATRIC.83332.12.NC_000962.CDS.1577604.1579580.fwd,PATRIC.83332.12.NC_000962.CDS.4081516.4082682.rev,PATRIC.83332.12.NC_000962.CDS.2052.3260.fwd,PATRIC.83332.12.NC_000962.CDS.3234189.3237806.rev,PATRIC.83332.12.NC_000962.CDS.4084453.4087257.rev,PATRIC.83332.12.NC_000962.CDS.5207.7267.fwd,PATRIC.83332.12.NC_000962.CDS.1937313.1938149.fwd,PATRIC.83332.12.NC_000962.CDS.34.1524.fwd,PATRIC.83332.12.NC_000962.CDS.4164995.4166731.rev,PATRIC.83332.12.NC_000962.CDS.624473.625480.rev,PATRIC.83332.12.NC_000962.CDS.1106405.1107382.rev,PATRIC.83332.12.NC_000962.CDS.524530.525390.rev,PATRIC.83332.12.NC_000962.CDS.1565441.1566697.fwd,PATRIC.83332.12.NC_000962.CDS.870008.871426.fwd,PATRIC.83332.12.NC_000962.CDS.1805653.1806000.fwd,PATRIC.83332.12.NC_000962.CDS.1801052.1802038.fwd,PATRIC.83332.12.NC_000962.CDS.4217134.4218195.fwd,PATRIC.83332.12.NC_000962.CDS.2993989.2995089.rev,PATRIC.83332.12.NC_000962.CDS.1795802.1797388.fwd,PATRIC.83332.12.NC_000962.CDS.3190701.3191621.rev,PATRIC.83332.12.NC_000962.CDS.219996.221723.rev,PATRIC.83332.12.NC_000962.CDS.1794786.1795805.fwd,PATRIC.83332.12.NC_000962.CDS.1564467.1565027.fwd,PATRIC.83332.12.NC_000962.CDS.1448094.1449371.fwd,PATRIC.83332.12.NC_000962.CDS.1201717.1203111.fwd,PATRIC.83332.12.NC_000962.CDS.4200388.4201326.fwd,PATRIC.83332.12.NC_000962.CDS.4153740.4155674)

then in IE Edge I get

TypeError: this.resolve is not a function
   at Function.lookup (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/express/lib/view.js:110:17)
   at View.lookup (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/enableMultipleViewRoots.js:22:25)
   at new View (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/express/lib/view.js:85:20)
   at EventEmitter.render (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/express/lib/application.js:569:12)
   at ServerResponse.render (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/express/lib/response.js:960:7)
   at WebFrameworkShim.applySegment (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/newrelic/lib/shim/shim.js:1257:17)
   at _applyRecorderSegment (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/newrelic/lib/shim/shim.js:938:24)
   at ServerResponse.render (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/newrelic/lib/shim/shim.js:918:17)
   at /disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/app.js:198:6
   at WebFrameworkShim.applySegment (/disks/space0/p3/production/p3-api/deployment/services/p3_api_service/app/node_modules/newrelic/lib/shim/shim.js:1257:17) 

in Chrome I get:

https://p3.theseed.org/services/data_api/genome_feature/?in(feature_id,(PATRIC.83332.12.NC_000962.CDS.2809936.2810928.rev,PATRIC.83332.12.NC_000962.CDS.2810993.2812066.rev,PATRIC.83332.12.NC_000962.CDS.4063254.4063856.rev,PATRIC.83332.12.NC_000962.CDS.1939599.1940291.fwd,PATRIC.83332.12.NC_000962.CDS.2412119.2413351.rev,PATRIC.83332.12.NC_000962.CDS.1674202.1675011.fwd,PATRIC.83332.12.NC_000962.CDS.2745314.2745736.rev,PATRIC.83332.12.NC_000962.CDS.3882834.3883442.fwd,PATRIC.83332.12.NC_000962.CDS.2516787.2517695.fwd,PATRIC.83332.12.NC_000962.CDS.4151180.4152196.rev,PATRIC.83332.12.NC_000962.CDS.7302.9818.fwd,PATRIC.83332.12.NC_000962.CDS.2710351.2711319.rev,PATRIC.83332.12.NC_000962.CDS.1938146.1938841.fwd,PATRIC.83332.12.NC_000962.CDS.1058260.1060575.fwd,PATRIC.83332.12.NC_000962.CDS.1577604.1579580.fwd,PATRIC.83332.12.NC_000962.CDS.4081516.4082682.rev,PATRIC.83332.12.NC_000962.CDS.2052.3260.fwd,PATRIC.83332.12.NC_000962.CDS.3234189.3237806.rev,PATRIC.83332.12.NC_000962.CDS.4084453.4087257.rev,PATRIC.83332.12.NC_000962.CDS.5207.7267.fwd,PATRIC.83332.12.NC_000962.CDS.1937313.1938149.fwd,PATRIC.83332.12.NC_000962.CDS.34.1524.fwd,PATRIC.83332.12.NC_000962.CDS.4164995.4166731.rev,PATRIC.83332.12.NC_000962.CDS.624473.625480.rev,PATRIC.83332.12.NC_000962.CDS.1106405.1107382.rev,PATRIC.83332.12.NC_000962.CDS.524530.525390.rev,PATRIC.83332.12.NC_000962.CDS.1565441.1566697.fwd,PATRIC.83332.12.NC_000962.CDS.870008.871426.fwd,PATRIC.83332.12.NC_000962.CDS.1805653.1806000.fwd,PATRIC.83332.12.NC_000962.CDS.1801052.1802038.fwd,PATRIC.83332.12.NC_000962.CDS.4217134.4218195.fwd,PATRIC.83332.12.NC_000962.CDS.2993989.2995089.rev,PATRIC.83332.12.NC_000962.CDS.1795802.1797388.fwd,PATRIC.83332.12.NC_000962.CDS.3190701.3191621.rev,PATRIC.83332.12.NC_000962.CDS.219996.221723.rev,PATRIC.83332.12.NC_000962.CDS.1794786.1795805.fwd,PATRIC.83332.12.NC_000962.CDS.1564467.1565027.fwd,PATRIC.83332.12.NC_000962.CDS.1448094.1449371.fwd,PATRIC.83332.12.NC_000962.CDS.1201717.1203111.fwd,PATRIC.83332.12.NC_000962.CDS.4200388.4201326.fwd,PATRIC.83332.12.NC_000962.CDS.4153740.4155674.fwd,PATRIC.83332.12.NC_000962.CDS.2718173.2718808.rev,PATRIC.83332.12.NC_000962.CDS.2862673.2863203.rev,PATRIC.83332.12.NC_000962.CDS.493851.495008.fwd,PATRIC.83332.12.NC_000962.CDS.525387.526082.rev,PATRIC.83332.12.NC_000962.CDS.1835013.1836236.fwd,PATRIC.83332.12.NC_000962.CDS.2379806.2380660.rev,PATRIC.83332.12.NC_000962.CDS.3359585.3360598.rev,PATRIC.83332.12.NC_000962.CDS.3179368.3180513.rev,PATRIC.83332.12.NC_000962.CDS.2416396.2417475.rev,PATRIC.83332.12.NC_000962.CDS.826173.826667.fwd,PATRIC.83332.12.NC_000962.CDS.1806181.1807263.fwd,PATRIC.83332.12.NC_000962.CDS.2861588.2862676.rev,PATRIC.83332.12.NC_000962.CDS.1136573.1138060.rev,PATRIC.83332.12.NC_000962.CDS.3157521.3158144.rev,PATRIC.83332.12.NC_000962.CDS.3073130.3073615.rev,PATRIC.83332.12.NC_000962.CDS.1872639.1874051.fwd,PATRIC.83332.12.NC_000962.CDS.3181770.3183059.rev,PATRIC.83332.12.NC_000962.CDS.1918746.1919669.fwd,PATRIC.83332.12.NC_000962.CDS.14914.15612.fwd,PATRIC.83332.12.NC_000962.CDS.1807903.1809453.fwd,PATRIC.83332.12.NC_000962.CDS.3073680.3074471.rev,PATRIC.83332.12.NC_000962.CDS.1799565.1800899.fwd,PATRIC.83332.12.NC_000962.CDS.1055024.1056685.rev,PATRIC.83332.12.NC_000962.CDS.4311704.4312675.rev,PATRIC.83332.12.NC_000962.CDS.1552654.1553235.fwd,PATRIC.83332.12.NC_000962.CDS.1449375.1450700.fwd,PATRIC.83332.12.NC_000962.CDS.3348805.3350277.rev,PATRIC.83332.12.NC_000962.CDS.1204067.1205233.fwd,PATRIC.83332.12.NC_000962.CDS.1470321.1471577.fwd,PATRIC.83332.12.NC_000962.CDS.3093905.3094879.rev,PATRIC.83332.12.NC_000962.CDS.2686367.2687131.fwd,PATRIC.83332.12.NC_000962.CDS.3318330.3318815.rev,PATRIC.83332.12.NC_000962.CDS.2380663.2380944.rev,PATRIC.83332.12.NC_000962.CDS.398658.399524.fwd,PATRIC.83332.12.NC_000962.CDS.305825.306349.rev,PATRIC.83332.12.NC_000962.CDS.2474864.2475970.rev,PATRIC.83332.12.NC_000962.CDS.2134273.2134872.rev,PATRIC.83332.12.NC_000962.CDS.1109272.1110192.fwd,PATRIC.83332.12.NC_000962.CDS.2767671.2768171.rev,PATRIC.83332.12.NC_000962.CDS.763370.767320.fwd,PATRIC.83332.12.NC_000962.CDS.3239829.3240551.rev,PATRIC.83332.12.NC_000962.CDS.3877464.3878507.rev,PATRIC.83332.12.NC_000962.CDS.3017829.3019421.fwd,PATRIC.83332.12.NC_000962.CDS.4410412.4410600.rev,PATRIC.83332.12.NC_000962.CDS.760035.763325.fwd,PATRIC.83332.12.NC_000962.CDS.3148385.3149428.rev,PATRIC.83332.12.NC_000962.CDS.2871206.2872015.rev,PATRIC.83332.12.NC_000962.CDS.2734792.2737008.rev,PATRIC.83332.12.NC_000962.CDS.3628160.3629647.rev,PATRIC.83332.12.NC_000962.CDS.2733299.2734144.fwd,PATRIC.83332.12.NC_000962.CDS.2455631.2456743.rev,PATRIC.83332.12.NC_000962.CDS.644499.645470.fwd,PATRIC.83332.12.NC_000962.CDS.4023868.4024269.rev,PATRIC.83332.12.NC_000962.CDS.1803294.1804031.fwd,PATRIC.83332.12.NC_000962.CDS.3431979.3433622.rev,PATRIC.83332.12.NC_000962.CDS.1566825.1568036.fwd,PATRIC.83332.12.NC_000962.CDS.2392517.2393320.rev,PATRIC.83332.12.NC_000962.CDS.645467.646255.fwd,PATRIC.83332.12.NC_000962.CDS.1451997.1452947.fwd,PATRIC.83332.12.NC_000962.CDS.1130155.1131111.fwd,PATRIC.83332.12.NC_000962.CDS.435471.436769.rev,PATRIC.83332.12.NC_000962.CDS.1582166.1583104.fwd,PATRIC.83332.12.NC_000962.CDS.2863207.2864445.rev,PATRIC.83332.12.NC_000962.CDS.3299971.3300609.rev,PATRIC.83332.12.NC_000962.CDS.801462.802133.fwd,PATRIC.83332.12.NC_000962.CDS.800809.801462.fwd,PATRIC.83332.12.NC_000962.CDS.4307655.4308914.rev,PATRIC.83332.12.NC_000962.CDS.3093479.3093748.rev,PATRIC.83332.12.NC_000962.CDS.2739772.2740032.rev,PATRIC.83332.12.NC_000962.CDS.2896013.2897803.rev,PATRIC.83332.12.NC_000962.CDS.3199119.3199967.rev,PATRIC.83332.12.NC_000962.CDS.1446379.1448031.fwd,PATRIC.83332.12.NC_000962.CDS.802133.802435.fwd,PATRIC.83332.12.NC_000962.CDS.805110.805526.fwd,PATRIC.83332.12.NC_000962.CDS.1455262.1455405.fwd,PATRIC.83332.12.NC_000962.CDS.3217155.3217643.rev,PATRIC.83332.12.NC_000962.CDS.802528.803370.fwd,PATRIC.83332.12.NC_000962.CDS.3879696.3880070.rev,PATRIC.83332.12.NC_000962.CDS.812976.813374.fwd,PATRIC.83332.12.NC_000962.CDS.58192.58482.fwd,PATRIC.83332.12.NC_000962.CDS.805526.805759.fwd,PATRIC.83332.12.NC_000962.CDS.3861947.3862390.rev,PATRIC.83332.12.NC_000962.CDS.3878659.3879264.rev,PATRIC.83332.12.NC_000962.CDS.4041423.4042940.rev,PATRIC.83332.12.NC_000962.CDS.814993.815190.fwd,PATRIC.83332.12.NC_000962.CDS.3367264.3368793.rev,PATRIC.83332.12.NC_000962.CDS.1736519.1739644.fwd,PATRIC.83332.12.NC_000962.CDS.1859758.1862253.fwd,PATRIC.83332.12.NC_000962.CDS.1858700.1859758.fwd,PATRIC.83332.12.NC_000962.CDS.1913602.1914876.fwd,PATRIC.83332.12.NC_000962.CDS.803411.803692.fwd,PATRIC.83332.12.NC_000962.CDS.3879273.3879692.rev,PATRIC.83332.12.NC_000962.CDS.800487.800792.fwd,PATRIC.83332.12.NC_000962.CDS.811373.811741.fwd,PATRIC.83332.12.NC_000962.CDS.805798.806166.fwd,PATRIC.83332.12.NC_000962.CDS.748276.748812.fwd,PATRIC.83332.12.NC_000962.CDS.2904821.2906092.rev,PATRIC.83332.12.NC_000962.CDS.4063901.4064785.rev,PATRIC.83332.12.NC_000962.CDS.2873771.2876485.rev,PATRIC.83332.12.NC_000962.CDS.781560.781934.fwd,PATRIC.83332.12.NC_000962.CDS.1125444.1127003.rev,PATRIC.83332.12.NC_000962.CDS.3861495.3861950.rev,PATRIC.83332.12.NC_000962.CDS.59397.59867.fwd,PATRIC.83332.12.NC_000962.CDS.803689.804282.fwd,PATRIC.83332.12.NC_000962.CDS.813398.813937.fwd,PATRIC.83332.12.NC_000962.CDS.2740047.2740361.rev,PATRIC.83332.12.NC_000962.CDS.811742.812059.fwd,PATRIC.83332.12.NC_000962.CDS.812059.812622.fwd,PATRIC.83332.12.NC_000962.CDS.735517.736224.fwd,PATRIC.83332.12.NC_000962.CDS.3876890.3877432.rev,PATRIC.83332.12.NC_000962.CDS.43562.46471.fwd,PATRIC.83332.12.NC_000962.CDS.3369950.3371434.rev,PATRIC.83332.12.NC_000962.CDS.1853184.1853573.fwd,PATRIC.83332.12.NC_000962.CDS.781934.782404.fwd,PATRIC.83332.12.NC_000962.CDS.814328.814990.fwd,PATRIC.83332.12.NC_000962.CDS.1094886.1095059.fwd,PATRIC.83332.12.NC_000962.CDS.2941189.2943564.rev,PATRIC.83332.12.NC_000962.CDS.3722621.3723631.rev,PATRIC.83332.12.NC_000962.CDS.3371431.3371730.rev,PATRIC.83332.12.NC_000962.CDS.804282.805106.fwd,PATRIC.83332.12.NC_000962.CDS.3880286.3880399.rev,PATRIC.83332.12.NC_000962.CDS.1852928.1853122.fwd,PATRIC.83332.12.NC_000962.CDS.815190.815630.fwd,PATRIC.83332.12.NC_000962.CDS.2747595.2750255.rev,PATRIC.83332.12.NC_000962.CDS.3213912.3214253.rev,PATRIC.83332.12.NC_000962.CDS.813940.814308.fwd,PATRIC.83332.12.NC_000962.CDS.2710075.2710335.fwd,PATRIC.83332.12.NC_000962.CDS.4022394.4023803.rev,PATRIC.83332.12.NC_000962.CDS.1802664.1803284.fwd,PATRIC.83332.12.NC_000962.CDS.1804853.1805656.fwd,PATRIC.83332.12.NC_000962.CDS.1214049.1214360.fwd,PATRIC.83332.12.NC_000962.CDS.3352458.3353486.rev,PATRIC.83332.12.NC_000962.CDS.1797388.1798245.fwd,PATRIC.83332.12.NC_000962.CDS.2410638.2412122.rev,PATRIC.83332.12.NC_000962.CDS.1068205.1069776.fwd,PATRIC.83332.12.NC_000962.CDS.1871363.1872559.fwd,PATRIC.83332.12.NC_000962.CDS.4024344.4025039.rev,PATRIC.83332.12.NC_000962.CDS.3414232.3414684.rev,PATRIC.83332.12.NC_000962.CDS.3412085.3414166.rev,PATRIC.83332.12.NC_000962.CDS.3840194.3841420.rev,PATRIC.83332.12.NC_000962.CDS.3361130.3362986.rev,PATRIC.83332.12.NC_000962.CDS.3360624.3361130.rev,PATRIC.83332.12.NC_000962.CDS.1450697.1451779.fwd,PATRIC.83332.12.NC_000962.CDS.570539.571648.fwd,PATRIC.83332.12.NC_000962.CDS.2682015.2683187.rev,PATRIC.83332.12.NC_000962.CDS.3592254.3593372.fwd,PATRIC.83332.12.NC_000962.CDS.2472493.2473242.fwd,PATRIC.83332.12.NC_000962.CDS.2992634.2993947.rev,PATRIC.83332.12.NC_000962.CDS.1135501.1136481.rev,PATRIC.83332.12.NC_000962.CDS.2684703.2686370.fwd,PATRIC.83332.12.NC_000962.CDS.1030578.1031864.rev,PATRIC.83332.12.NC_000962.CDS.1381942.1383042.rev,PATRIC.83332.12.NC_000962.CDS.168704.170182.rev,PATRIC.83332.12.NC_000962.CDS.3603425.3604729.fwd,PATRIC.83332.12.NC_000962.CDS.2861148.2861591.rev,PATRIC.83332.12.NC_000962.CDS.646298.647962.fwd,PATRIC.83332.12.NC_000962.CDS.2440332.2441720.rev,PATRIC.83332.12.NC_000962.CDS.2382489.2386067.rev,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined))&limit(25000)&http_authorization=un%3Dfbjornlund%40patricbrc.org%7Ctokenid%3D632d44e7-d28c-425c-b4e1-192882672e90%7Cexpiry%3D1537343991%7Cclient_id%3Dfbjornlund%40patricbrc.org%7Ctoken_type%3DBearer%7Crealm%3Dpatricbrc.org%7CSigningSubject%3Dhttps%3A%2F%2Fuser.patricbrc.org%2Fpublic_key%7Csig%3D7a04a6de0cd35b375690305539441e17dac6ef87564d42af2bae6f9a6844620354b1d38541ce3e0c55c02ceaa6b1e13a26930d1c2875c346fdb43fd92380481ba736dd3240b89fcc3b5bf11e0f3714367373e44f6756465313facd5af06ac6baee7c24948fb8cbf2769ec1d44fd3485a92f9a9b78f1ebc17dc238882f480906c&http_accept=application/dna+fasta

414 Request-URI Too Large

The edge url seems to be truncated relative to chrome url.

See https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers for guidelines on url usage. Patrics huge urls seems to be bad practise.

**Comment by dawenx** [comment]
The problem is caused by rows in 2nd page and I did not load IDs, e.g., those "undefined" in the long internal URL. With help from @dmachi , I got this issue fixed. @hyoo , can you confirm it and push it to alpha if the problem is fixed. We do have a way to select data for all rows in the grids, but I did not enable it when I worked on issue #1452.

@fbjornlund , thanks for reporting the issue and providing detailed information. How did you generate the complete link in Chrome? Is it copied from debugger or it's from our UI, if so, could you please let us know how to reproduce it? We did handle long URL internally.

**Comment by fbjornlund** [comment]
Thanks for moving fast towards a solution, I hope to soon see it in production :)

I copied the url from the chrome browsers adress bar, no debugger or special tool. I tried to reproduce it again, and I got the same behavior. 1) Select all (multiple pages) 2) Open fasta viewer (creates a new tab, display of sequence fails) 3) Click on download - this opens a new tab with the long ur in the adress bar, and '414 Request-URI Too Large' in the tab title and displayed in the content area in the browser.

I've attached a screenshot displaying the behavior. The two tabs to the left of the active tab is the tabs for the fasta viewer, and for the tabular display of the speciality genes.

![patric_long_url](https://user-images.githubusercontent.com/7874477/30609013-7134dec0-9d7a-11e7-85ce-874199468edd.png)

Whats the normal time frame for making a solution like this available to users?

**Comment by dawenx** [comment]
Thanks @fbjornlund , I was able to reproduce the 414 request issue, this seems to be a difference problem and we will look into it.

Once the fix I did earlier was deployed to the production, you should be able to download fasta data directly from the results pages by: 1) Select all 2) Click on 'DWNLD' in the green action bar (above where you chose 'FASTA viewer').

We will let you know when 414 request issue is fixed.

**Comment by hyoo** [comment]
There are several issues in this ticket.
1. @fbjornlund 414 error is due to the length of URL we have generated. Copying and paste the url above will not work ever, since this url length is limited by standard. We need to move this request to use POST method. I guess @dawenx already know how to do it.

2. picking rows over multiple page is a site-wide issue. It has some complicated issue and we need to address it asap. I think @dmachi knows in detail. Dustin, If you can help @dawenx to understand the problem, so that he can start working on it, it will be the best.

3. downloading fasta will be handled by data api. it doesn't matter whether the request comes from features grid or sp grid, as long as it contains feature_ids.

**Comment by dawenx** [comment]
@hyoo just wanted to be clear: 2 was addressed in PR 545 that you just merged. Yes, @dmachi has provided a way to address multiple page data loading issue. We probably need to go through the site to add this. 3 is not a problem, just a FYI, right?

I will work 1 and yes, we need to use POST.

**Comment by hyoo** [comment]
![image](https://user-images.githubusercontent.com/1236467/30610905-57a08de0-9d45-11e7-864f-156a44debc39.png)

I will be surprised this one line address all the issues. I have tested on alpha, the problem still exists.

lets separate issues. I will create another ticket and will link all the issues related.

**Comment by dawenx** [comment]
@hyoo yes, one-line... @dmachi did the heavy listing in PageGrid.js, all we need is to enable it by specifying the selectAllFields and it only addressed issue 2 in your summary.

**Comment by dawenx** [comment]
@hyoo i checked out FASTA.js, it's using 'post' there, not sure why it behaves like using get. could you please take a look? i will work on the new ticket you just created and will try to go through all grids later on to apply this fix for multiple page data loading issue.

**Comment by dawenx** [comment]
@hyoo I took it back on handling issue 2 already, in stead, loading all data is possible with PageGrid.js.

**Comment by dawenx** [comment]
I fixed the 2nd problem reported in this issue -- download bug in FASTA viewer. @hyoo I have submitted PR on this. This issue can be close after you verify it. I will work on selection on multiple pages issue.
@fbjornlund Hopefully my fix will be on the production soon. If you want a particular file from the download, I can upload it there while we are waiting for the codes to be rolled out. Thanks for your patience,
